src/rust/lib/debug-scenes/src/ports.rs

```diff
@@ -0,0 +1,99 @@
+#![allow(missing_docs)]
+
+//! NOTE
+//! This file is under a heavy development. It contains commented lines of code and some code may
+//! be of poor quality. Expect drastic changes.
+
+use ensogl::traits::*;
+
+use ensogl::data::color::*;
+use ensogl::display::shape::*;
+use ensogl::display::shape::compound::port::{PortSpecification, Port, PortDirection};
+use ensogl::display::shape::primitive::system::ShapeSystem;
+use ensogl::display::symbol::geometry::Sprite;
+use ensogl::display::world::*;
+use ensogl::math::topology::unit::AngleOps;
+use ensogl::system::web;
+use nalgebra::Vector2;
+use wasm_bindgen::prelude::*;
+
+
+
+#[wasm_bindgen]
+#[allow(dead_code)]
+pub fn run_example_ports() {
+    web::forward_panic_hook_to_console();
+    web::set_stdout();
+    web::set_stack_trace_limit();
+    init(&WorldData::new(&web::get_html_element_by_id("root").unwrap()));
+}
+
+
+fn init(world: &World) {
+    let scene  = world.scene();
+    let camera = scene.camera();
+    let screen = camera.screen();
+
+    let port_spec = PortSpecification{
+        height: 30.0,
+        width: Angle::from(45.0),
+        inner_radius: 80.0,
+        direction: PortDirection::Outwards,
+        location: 90.0_f32.deg(),
+        color: Srgb::new(51.0 / 255.0, 102.0 / 255.0, 153.0 / 255.0 ),
+    };
+
+    let port_shape_1 = Port::new(port_spec);
+
+    let port_spec = PortSpecification{
+        height: 30.0,
+        width: Angle::from(45.0),
+        inner_radius: 80.0,
+        direction: PortDirection::Inwards,
+        location: 270.0_f32.deg(),
+        color: Srgb::new(51.0 / 255.0, 102.0 / 255.0, 153.0 / 255.0 ),
+    };
+
+    let port_shape_2 = Port::new(port_spec);
+
+    let shape_system =     ShapeSystem::new(world,&port_shape_1);
+    let sprite = shape_system.new_instance();
+    sprite.size().set(Vector2::new(500.0,500.0));
+    sprite.mod_position(|t| {
+        t.x += screen.width / 2.0;
+        t.y += screen.height / 2.0;
+    });
```

Comment:
This sprite, display object etc should be part of "Port" structure. `Port` should be jsut an object representing port on the stage.